pandas/core/frame.py

```diff
@@ -4533,6 +4533,14 @@ def dropna(self, axis=0, how='any', thresh=None, subset=None,
             agg_obj = self
             if subset is not None:
                 ax = self._get_axis(agg_axis)
+                if isinstance(ax, MultiIndex):
+                    if (len(subset[0]) < ax.nlevels and
+                       isinstance(subset[0], tuple)):
+                        for _ in range(ax.nlevels - len(subset[0])):
+                            ax = ax.droplevel(ax.nlevels - 1)
```

Comment:
you should be able to just change this to ``.get_loc`` and it should work; the return value can be different though

```
In [1]: mi = pd.MultiIndex.from_product([list('abc'), list(range(3))])                                                                                                                                                                  

In [2]: mi.get_loc('a')                                                                                                                                                                                                                 
Out[2]: slice(0, 3, None)

In [3]: i = pd.Index(range(3))                                                                                                                                                                                                          

In [4]: i.get_loc(2)                                                                                                                                                                                                                    
Out[4]: 2
```

Reply:
though you may not need to do any of this, just

```
agg_obj = self.loc(axis=axis)[subset]
```

Reply:
Thank you for the feedback, I will take a look. Sorry for the late reply.

Reply:
@jreback I've been trying to play around with what you suggested, but so far I cannot get it to work. I will keep trying, but wanted to ask if you have any further suggestions?